subprojects/ear/src/main/java/org/gradle/plugins/ear/Ear.java

```diff
@@ -271,4 +275,18 @@ public void setDeploymentDescriptor(DeploymentDescriptor deploymentDescriptor) {
         this.deploymentDescriptor = deploymentDescriptor;
     }
 
+    /**
+     * The name of the application directory, relative to the project directory.. Added to the output ear archive by default.
+     * <p>
+     * The {@code ear} plugin sets the default value for all {@code Ear} tasks to {@code src/main/application}.
+     * <p>
+     * Note, that if the {@code ear} plugin is not applied then this property is ignored.
+     *
+     * @since 7.1
+     */
+    @Internal
+    @Incubating
+    public Property<String> getAppDirName() {
```

Comment:
it looks like this is a path like in the War task, so should it be a `DirectoryProperty` too?

Reply:
That's only because I didn't want to introduce any structural changes.

Reply:
Fixed.